apps/extension/src/background/keyring/service.ts

```diff
@@ -264,6 +265,44 @@ export class KeyRingService {
     }
   }
 
+  async submitClaimRewards(
+    claimRewardsMsg: string,
+    txMsg: string,
+    msgId: string
+  ): Promise<void> {
+    await this.broadcaster.startTx(msgId, TxType.ClaimRewards);
+    try {
+      const txMsgValue = Message.decode(fromBase64(txMsg), TxMsgValue);
+      const claimRewardsMsgValue = Message.decode(
+        fromBase64(claimRewardsMsg),
+        ClaimRewardsMsgValue
+      );
+
+      const innerTxHash = await this._keyRing.submitClaimRewards(
+        claimRewardsMsgValue,
+        txMsgValue
+      );
+      await this.broadcaster.completeTx(
+        msgId,
+        TxType.ClaimRewards,
+        true,
+        innerTxHash
+      );
+      // TODO: not sure if it is correct to broadcast staking update
```

Comment:
It looks like staking update only updates the list of validators, and AFAIK the rewards are not part of bonded stake so it should not be needed :)